### ISSUE TITLE
Linestring empty when intersecting vertical line with a square

### ISSUE THREAD

**Issue by Cactusbone** [libgeos/geos#700]
Hello, I've stumbled on something weird :

Using 
```sql
SELECT ST_Astext(
	ST_Intersection(
		ST_GeomFromText('LINESTRING(1 1, 1 2)'),
		ST_GeomFromText('POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))')
	)
)
```

returns `LINESTRING EMPTY`

on PostGIS 3.2 (`"POSTGIS=""3.2.3 2f97b6c"" [EXTENSION] PGSQL=""140"" GEOS=""3.9.0-CAPI-1.16.2"" SFCGAL=""1.3.8"" PROJ=""7.2.1"" GDAL=""GDAL 3.2.2, released 2021/03/05"" LIBXML=""2.9.10"" LIBJSON=""0.15"" LIBPROTOBUF=""1.3.3"" WAGYU=""0.5.0 (Internal)"" RASTER"` or `"POSTGIS=""3.2.1 5fae8e5"" [EXTENSION] PGSQL=""140"" GEOS=""3.9.0-CAPI-1.16.2"" SFCGAL=""1.3.8"" PROJ=""7.2.1"" GDAL=""GDAL 3.2.2, released 2021/03/05"" LIBXML=""2.9.10"" LIBJSON=""0.15"" LIBPROTOBUF=""1.3.3"" WAGYU=""0.5.0 (Internal)"" TOPOLOGY RASTER"`)

It works on PostGIS 2.5 (`"POSTGIS=""2.5.1 r17027"" [EXTENSION] PGSQL=""110"" GEOS=""3.7.1-CAPI-1.11.1 27a5e771"" SFCGAL=""1.3.6"" PROJ=""Rel. 5.2.0, September 15th, 2018"" GDAL=""GDAL 2.4.0, released 2018/12/14"" LIBXML=""2.9.4"" LIBJSON=""0.12.1"" LIBPROTOBUF=""1.2.1"" TOPOLOGY RASTER"`)

It's the same with horizontal lines
```sql
SELECT ST_Astext(
	ST_Intersection(
		ST_GeomFromText('LINESTRING(1 1, 2 1)'),
		ST_GeomFromText('POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))')
	)
)
```

However it works when the line is neither horizontal nor vertical or when it touches the intersection

Queries that work:
```sql
SELECT ST_Astext(ST_Intersection(
	ST_GeomFromText('LINESTRING(1 1, 2 2)'),
	ST_GeomFromText('POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))')
))
```
```sql
SELECT ST_Astext(ST_Intersection(
	ST_GeomFromText('LINESTRING(1 1, 2 1.000000000001)'),
	ST_GeomFromText('POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))')
))
```
```sql
SELECT ST_Astext(
	ST_Intersection(
		ST_GeomFromText('LINESTRING(0 0, 0 1)'),
		ST_GeomFromText('POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))')
	)
)
```

EDIT: on second thought, this might not be related to GEOS ;)

**Comment by pramsey** [comment]
Certainly with a contemporary version of GEOS this issue does not exist. 
```
SELECT ST_Astext(
	ST_Intersection(
		ST_GeomFromText('LINESTRING(1 1, 1 2)'),
		ST_GeomFromText('POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))')
	)
)

      st_astext      
---------------------
 LINESTRING(1 1,1 2)
```
You might find it goes away with the latest patch release of 3.9. Please close when you are satisfied.

**Comment by dr-jts** [comment]
This is a bug that was fixed in GEOS 3.9.1:https://github.com/libgeos/geos/commit/acefb04aac1352add0449bc7e6f38af5093134ce

**Comment by Cactusbone** [comment]
Great thx ! I'll move away from official postgis docker images then ;)
